doc/plutus-core-spec/cardano/builtins4.tex

```diff
@@ -37,13 +43,111 @@ \subsubsection{Miscellaneous built-in functions}
+                                        \itobsLE(w,n) & \text{if $e=\mathtt{false}$}\\
+                                        \itobsBE(w,n) & \text{if $e=\mathtt{true}$}\\
+                                        \end{cases}$}
+                                        & Yes & \ref{note:itobs}\strut\\ \strut
+    \TT{byteStringToInteger} & $[\ty{bool}, \ty{bytestring}] $ \text{\: $ \to \ty{bytestring}$}
+                                        & $(e, [c_0, \ldots, c_{N-1}]) $ \text{\; $\mapsto \begin{cases}
+                                        \sum_{i=0}^{N-1}c_{i}256^i & \text{if $e=\mathtt{false}$}\\
+                                        \sum_{i=0}^{N-1}c_{i}256^{N-1-i} & \text{if $e=\mathtt{true}$}\\
+                                        \end{cases}$}
+                                        &  & \ref{note:bstoi}\\
     \hline
 \end{longtable}
 
+\note{Integer to bytestring conversion.}
+\label{note:itobs}
+The \texttt{integerToByteString} function converts non-negative integers to bytestrings.
+It takes three arguments:
+\begin{itemize}
+\item A boolean endianness flag $e$.
+\item An integer width argument $w$ with $0 \leq w < 8192$.
+\item The integer $n$ to be converted: it is required that $0 \leq n < 256^{8192} = 2^{65536}$.
+\end{itemize}
+
+\noindent The conversion is little-endian if $e$ is \texttt{(con bool False)} and
+big-endian if $e$ is \texttt{(con bool True)}. If the width $w$ is zero then the
+output is a bytestring which is just large enough to hold the converted integer.
+If $w>0$ then the output is exactly $w$ bytes long, and it is an error if $n$
+does not fit into a bytestring of that size; if necessary, the output is padded
+with \texttt{0x00} bytes (on the right in the little-endian case and the left in
+the big-endian case) to make it the correct length.  For example, the five-byte
+little-endian representation of the integer \texttt{0x123456} is the
+bytestring \texttt{[0x56, 0x34, 0x12, 0x00, 0x00]} and the five-byte big-endian
+representation is \texttt{[0x00, 0x00, 0x12, 0x34, 0x56]}.  In all cases an
+error occurs error if $w$ or $n$ lies outside the expected range, and in
+particular if $n$ is negative.
+
+\newpage
+\noindent The precise semantics of \texttt{integerToByteString} are given
+by the functions $\itobsLE: \Z \times \Z \rightarrow \withError{\B^*}$ and $\itobsBE
+: \Z \times \Z \rightarrow \withError{\B^*}$.  Firstly we deal with out-of-range cases and
+the case $n=0$:
+
+$$
+\itobsLE (w,n) = \itobsBE (w,n) = 
+\begin{cases}
+  \errorX & \text{if $n<0$ or $n \geq 2^{65536}$}\\
+  \errorX & \text{if $w<0$ or $w > 8192$}\\
+  [] & \text{if $n=0$ and $0 \leq w \leq 8192$}\\
+\end{cases}
+$$
+
+\noindent  Now assume that none of the conditions above hold, so $0 < n < 2^{65536}$ and
+$0 \leq w \leq 8192$.  Since $n>0$ we can write
+$n = \sum_{i=0}^{N-1}a_{i}256^i$ with $N \geq 1$ and $a_{N-1} \ne 0$.   We then  have
```

Comment:
Sorry, maybe it's a stupid question, but I don't understand why we can write `n` as that sum. Can you provide some more explanations?

Reply:
I guess it's a base-256 expansion: I'll try to explain that a bit more clearly.

Reply:
Oh, I forgot to say that the a_i are supposed to lie between 0 and 255. I've fixed that now.